w06-ruby-on-rails/d3-dawn-ruby-oop/README.md

```diff
@@ -0,0 +1,154 @@
+#Ruby <3's OOP
+
+##Learning Objectives
+By the end of lesson, students will be able to...
+
+* Distinguish between objects & classes in Ruby
+* Initialize classes
+* Define attributes & instance methods
+* Distinguish between instance and class variables
+
+##Hashes (10m)
+
+Let's create a new Hash
+
+* Hashes are simple key value stores.
+
+**Challenge:**
+How can I organize my data using key/value pairs in Ruby? Like so:
+
+```ruby
+{:name=>"Napoleon", :fav_food=>"steak", :skills=>["archery", "combat", "egg farming"]}
+```
+
+##Objects (10m)
+
+* Everything in Ruby is an Object; however, we almost never use plain vanilla Objects because there are more sophisticated implementations of them such a `String`, `Integer`, and `Hash`.
+
+**Challenge:**
+How can we prove that the Hash we just created inherited from `Basic Object`?
+
+###Class Inheritance Tree
+
+![Class inheritance](http://i.stack.imgur.com/rvcEi.png)
+
+##Classes (10m)
+
+**Challenge:**
+How do we create a class in Ruby?
+
+Goal: Let's create a Car that goes "Vroom" when it's first *initialized*
+
+*Refresher: Classes are datatypes used to create more data. They are analogous to constructors in JavaScript.*
+
+##Attributes (10m)
+
+What should we do if we want to set attributes on the car, such as a paint color and year?
+
+**Challenge:**
+Enable this code...
+
+```ruby
+fiat = Car.new
+fiat.color = "hot pink"
+fiat.color
+=> "hot pink"
+```
+
+*Hint: Use `attr_accessor`.*
```

Comment:
Has attr_accessor and the difference between methods and variables in an instance been explained at this point?  (Maybe that's part of the lecture that I don't see)  It might be confusing to them as its different than js.